### ISSUE TITLE
Trainer.hyperparameter_search is much faster than AutoML.fit when using the same config

### ISSUE THREAD

**Issue by liususan091219** [microsoft/FLAML#636]
Figure out the reason why.

**Comment by liususan091219** [comment]
After debugging, found that loss.backward for FLAML takes 100 times longer running time than transformers. 

FLAML's loss.backward takes 0.173, HF's loss.backward takes 0.02 secs.
